### ISSUE TITLE
Enhancement: [consistent-type-exports] Check * exports to see if all exported values are types

### ISSUE THREAD

**Issue by NotWoods** [typescript-eslint/typescript-eslint#9986]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/consistent-type-exports/

### Description

Currently, consistent-type-exports only checked named exports and re-exports. However, sometimes we might write a file that only contains types & interfaces. It would be good to enforce using `export type *` instead of `export *` if all the exports are types.

### Fail

```TypeScript
// Component.props.ts
export interface Props {}

// Component.ts
export function Component(props: Props) {}

// index.ts
export * from './Component.props'
export * from './Component'
```

### Pass

```TypeScript
// Component.props.ts
export interface Props {}

// Component.ts
export function Component(props: Props) {}

// index.ts
export type * from './Component.props' // changed
export * from './Component'
```

**Comment by JoshuaKGoldberg** [comment]
👍 from me on this one, seems like a fairly straightforward omission.

**Comment by abrahamguo** [comment]
@kirkwaiblinger @JoshuaKGoldberg

For the current functionality, we are iterating through each `Identifier` of an export, and using `checker.getAliasedSymbol(services.getSymbolAtLocation(Identifier))` to automatically resolve types when re-exporting from another file:

https://github.com/typescript-eslint/typescript-eslint/blob/ef3384e06028eda0fc6aef641f1fa93618f451f4/packages/eslint-plugin/src/rules/consistent-type-exports.ts#L88-L105

However, if we are to support `export * from ...`, there is no `Identifier` that we can give to TypeScript to request the type.
Thus, does this mean that we will need to manually resolve the export path ourselves?

**Comment by JoshuaKGoldberg** [comment]
We can always flag the `*` or `export *` an name the export explicitly.

**Comment by bradzacher** [comment]
I would think that there's a way for us to ask TS for the signature of the module so that we can inspect the things and determine if any of the exported names are values.
